### PR TITLE
Poll: Fix deletequeue

### PR DESCRIPTION
**Pull request by Distrib-ps**
Pr for fix /deletequeue
recording

### CODE REVIEW

File: server/chat-plugins/poll.ts [hunk 1]
```diff
@@ -573,7 +578,7 @@ export const pages: PageTable = {
 			const number = i + 1; // for translation convienence
 			const button = (
 				`<strong>${this.tr`#${number} in queue`} </strong>` +
-				`<button class="button" name="send" value="/poll deletequeue ${i + 1},${room.roomid},updatelist">` +
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:576]: Why? This is important, don't remove this.

File: server/chat-plugins/poll.ts [hunk 2]
```diff
@@ -350,27 +350,33 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
-			room = this.requireRoom();
-			this.checkCan('mute', null, room);
-			if (!room.minorActivityQueue) {
+			target = target.trim();
+			const [slotString, roomid] = target.split(',');
+			const Room = roomid ? Rooms.search(roomid) : room;
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:355]: Should be `room`, not Room.

Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:355]: This is additionally not the fix you want. This does not work.
This is what you want instead.
```ts
const [slotString, roomid, update] = target.split(',');
room = roomid ? Rooms.search(roomid) : room;
room = this.requireRoom();

```

None of these other changes are necessary.

**Review by mia-pi-git — CHANGES_REQUESTED** [review]
This does not work as is, and in fact breaks more functionality.

File: server/chat-plugins/poll.ts (hunk 2 above)
Comment (Distrib-ps) [reply server/chat-plugins/poll.ts:355]: If I add only this code, there is an error: http://prntscr.com/w99uye

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:355]: I mean that's pretty simple to fix.
Just make it into:
room = roomid ? Rooms.search(roomid) : room;
this.room = room;
room = this.requireRoom();

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts:355]: Sorry I'm trying to learn at the same time

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts:355]:
```suggestion
					clearqueue: 'deletequeue',
		deletequeue(target, room, user, connection, cmd) {
			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
			const targetRoom = roomid ? Rooms.search(roomid) : room;
			if (!targetRoom) return this.errorReply(`Invalid room.`);
			if (!targetRoom.persist) {
				return this.errorReply("This command is unavailable in temporary rooms.");
			}
			this.room = targetRoom;
			room = this.requireRoom();
			this.checkCan('mute', null, room);
			if (!targetRoom.minorActivityQueue) {
				return this.errorReply(this.tr`The queue is already empty.`);
			}
			if (cmd === 'deletequeue' && targetRoom.minorActivityQueue.length !== 1 && !target) {
				return this.parse('/help deletequeue');
			}
			if (!target) {
				targetRoom.minorActivityQueue = null;
				this.modlog('CLEARQUEUE');
				this.sendReply(this.tr`Cleared poll queue.`);
			} else {
				const [slotString, roomid, update] = target.split(',');
				const slot = parseInt(slotString);
				const curRoom = roomid ? (Rooms.search(roomid) as ChatRoom | GameRoom) : room;
				if (!curRoom) return this.errorReply(this.tr`Room "${roomid}" not found.`);
				if (isNaN(slot)) {
					return this.errorReply(this.tr`Can't delete poll at slot ${slotString} - "${slotString}" is not a number.`);
				}
				if (!targetRoom.minorActivityQueue[slot - 1]) return this.errorReply(this.tr`There is no poll in queue at slot ${slot}.`);

				curRoom.minorActivityQueue!.splice(slot - 1, 1);
				if (!curRoom.minorActivityQueue?.length) curRoom.minorActivityQueue = null;

				curRoom.modlog({
					action: 'DELETEQUEUE',
					loggedBy: user.id,
					note: slot.toString(),
				});
				curRoom.sendMods(this.tr`(${user.name} deleted the queued poll in slot ${slot}.)`);
				curRoom.update();
				if (update) this.parse(`/j view-pollqueue-${curRoom}`);
			}
		},
```
It's good like that ?

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:355]: Remove line 373. Duplicated code.

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:355]: You also need the "update" variable to update the page if it's there.

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:355]: At line 391 you check for if (roomid) (parse join), but that should actually be if (update), as well.

File: server/chat-plugins/poll.ts [hunk 3]
```diff
@@ -382,6 +388,7 @@ export const commands: ChatCommands = {
 				});
 				curRoom.sendMods(this.tr`(${user.name} deleted the queued poll in slot ${slot}.)`);
 				curRoom.update();
+				if (roomid) this.parse(`/j view-pollqueue-${curRoom}`);
```
Comment (AnnikaCodes) [inline server/chat-plugins/poll.ts:391]: What is this line for?

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts:391]: Oops it was for testing.
You know how to fix the 120 character limit problem ?

File: server/chat-plugins/poll.ts [hunk 4]
```diff
@@ -350,27 +350,33 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			const targetRoom = roomid ? Rooms.search(roomid) : room;
+			if (!targetRoom) return this.errorReply(`Invalid room.`);
+			if (!targetRoom.persist) {
+				return this.errorReply("This command is unavailable in temporary rooms.");
+			}
+			this.room = targetRoom;
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:359]: Additionally, set `room = targetRoom`, then change all the uses of `targetRoom` back to `room`. Cleaner that way.

File: server/chat-plugins/poll.ts [hunk 5]
```diff
@@ -350,27 +350,33 @@ export const commands: ChatCommands = {
...
 			} else {
-				const [slotString, roomid, update] = target.split(',');
 				const slot = parseInt(slotString);
 				const curRoom = roomid ? (Rooms.search(roomid) as ChatRoom | GameRoom) : room;
 				if (!curRoom) return this.errorReply(this.tr`Room "${roomid}" not found.`);
 				if (isNaN(slot)) {
 					return this.errorReply(this.tr`Can't delete poll at slot ${slotString} - "${slotString}" is not a number.`);
 				}
-				if (!room.minorActivityQueue[slot - 1]) return this.errorReply(this.tr`There is no poll in queue at slot ${slot}.`);
+				if (!targetRoom.minorActivityQueue[slot - 1]) return this.errorReply(this.tr`There is no poll in queue at slot ${slot}.`);
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:379]:
```suggestion
				if (!room.minorActivityQueue[slot - 1]) return this.errorReply(this.tr`There is no poll in queue at slot ${slot}.`);
```

Changing this back to `room` like I said above will fix the max-len error,

**Comment by Distrib-ps** [comment]
I put a message here.
I don't know if she was forgotten. Thx

File: server/chat-plugins/poll.ts [hunk 6]
```diff
@@ -350,6 +350,14 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			const targetRoom = roomid ? Rooms.search(roomid) : room;
+			if (!targetRoom) return this.errorReply(`Invalid room.`);
+			if (!targetRoom.persist) {
+				return this.errorReply("This command is unavailable in temporary rooms.");
+			}
+			this.room = targetRoom;
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:359]:
```suggestion
			this.room = room = targetRoom;
```
Probably cleaner.

File: server/chat-plugins/poll.ts [hunk 7]
```diff
@@ -350,6 +350,14 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			const targetRoom = roomid ? Rooms.search(roomid) : room;
+			if (!targetRoom) return this.errorReply(`Invalid room.`);
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:355]: Not necessary since requireRoom below does that.

File: server/chat-plugins/poll.ts [hunk 8]
```diff
@@ -350,6 +350,14 @@ export const commands: ChatCommands = {
...
+			if (!targetRoom.persist) {
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:356]: Do this below the requireRoom call.

**Review by mia-pi-git — CHANGES_REQUESTED** [review]

**Comment by Distrib-ps** [comment]
Like this?

File: server/chat-plugins/poll.ts [hunk 9]
```diff
@@ -350,6 +350,11 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			const targetRoom = roomid ? Rooms.search(roomid) : room;
+			if (!targetRoom) return this.errorReply(`Invalid room.`);
+			this.room = room = targetRoom;
+			room = targetRoom;
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:357]: You can remove this line.

File: server/chat-plugins/poll.ts [hunk 10]
```diff
@@ -363,7 +367,6 @@ export const commands: ChatCommands = {
 				this.modlog('CLEARQUEUE');
 				this.sendReply(this.tr`Cleared poll queue.`);
 			} else {
-				const [slotString, roomid, update] = target.split(',');
 				const slot = parseInt(slotString);
 				const curRoom = roomid ? (Rooms.search(roomid) as ChatRoom | GameRoom) : room;
 				if (!curRoom) return this.errorReply(this.tr`Room "${roomid}" not found.`);
```
Comment (mia-pi-git) [inline server/chat-plugins/poll.ts:372]: This can be removed. Same opts are done above.

File: server/chat-plugins/poll.ts (hunk 7 above)
Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:355]: This check if targetRoom is falsy is still unnecessary - requireRoom does the same thing below.

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts]: Yep, Done !

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts]: Removing the lines, the deleteqeue does not work. @mia-pi-git

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts]: How so?

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts]: Oh. you removed too much of the code.
Restore the targetRoom stuff, you should only have removed the `if (!targetRoom)` line.

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts]:
```ts
if (roomid) {
const targetRoom = Rooms.search(target);
if (targetRoom) {
this.room = targetRoom;
}
}
room = this.requireRoom(); // etc
```

Comment (Distrib-ps) [reply server/chat-plugins/poll.ts]: Done, thx !

**Review by mia-pi-git — APPROVED** [review]
Looks like that's everything from me.

File: server/chat-plugins/poll.ts [hunk 11]
```diff
@@ -350,6 +350,13 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			if (roomid) {
+				const targetRoom = roomid ? Rooms.search(roomid) : room;
+				if (targetRoom) {
+					this.room = targetRoom;
```
Comment (AnnikaCodes) [inline server/chat-plugins/poll.ts:357]: Wouldn't setting `this.room` make it difficult to tell if the command succeeded or failed, since its output would be in a different room than it was used in? Or is this command only used from the poll management page? (I dislike using pages personally, and prefer commands. I don't know if others share this preference.)

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:357]: This is meant for the page, yes.

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:357]: `this.room` has to be set for it to work.

File: server/chat-plugins/poll.ts [hunk 12]
```diff
@@ -363,10 +370,8 @@ export const commands: ChatCommands = {
 				this.modlog('CLEARQUEUE');
 				this.sendReply(this.tr`Cleared poll queue.`);
 			} else {
-				const [slotString, roomid, update] = target.split(',');
 				const slot = parseInt(slotString);
 				const curRoom = roomid ? (Rooms.search(roomid) as ChatRoom | GameRoom) : room;
-				if (!curRoom) return this.errorReply(this.tr`Room "${roomid}" not found.`);
```
Comment (Zarel) [inline server/chat-plugins/poll.ts:369]: Why was this deleted? This will cause a crash if it doesn't exist, won't it?

Comment (mia-pi-git) [reply server/chat-plugins/poll.ts:369]: we really shouldn't be using `curRoom` at all, `room` is updated to the new room above.

File: server/chat-plugins/poll.ts (hunk 11 above)
Comment (Zarel) [reply server/chat-plugins/poll.ts:357]: I'm pretty sure `this.room` does not actually need ot be set for it to work.

File: server/chat-plugins/poll.ts (hunk 12 above)
Comment (Zarel) [reply server/chat-plugins/poll.ts:369]: You approved this... :/
